src/main/java/org/sonarsource/sonarlint/ls/CommandManager.java

```diff
@@ -191,14 +193,33 @@ private void computeCodeActionsForSonarLintIssues(Diagnostic diagnostic, List<Ei
     }
   }
 
+  private Optional<CodeAction> createResolveIssueCodeAction(Diagnostic diagnostic, URI uri, ProjectBindingWrapper binding, String ruleKey,
+    IssuesCache.VersionedIssue versionedIssue) {
+    var isDelegatingIssue = versionedIssue.getIssue() instanceof DelegatingIssue;
+    var delegatingIssue = isDelegatingIssue ? ((DelegatingIssue) versionedIssue.getIssue()) : null;
+    if (delegatingIssue != null && delegatingIssue.getIssueId() != null) {
+      var issueId = delegatingIssue.getIssueId();
+      var serverIssueKey = delegatingIssue.getServerIssueKey();
+      var key = serverIssueKey == null ? issueId.toString() : serverIssueKey;
+      var changeStatusPermittedResponse =
+        Utils.safelyGetCompletableFuture(backendServiceFacade.checkChangeIssueStatusPermitted(
+          new CheckStatusChangePermittedParams(binding.getConnectionId(), key)
+        ));
+      if (changeStatusPermittedResponse.isPresent() && changeStatusPermittedResponse.get().isPermitted()) {
+        return Optional.of(createResolveIssueCodeAction(diagnostic, ruleKey, key, uri, false));
```

Comment:
Why is last parameter - `isTaintIssue` automatically `false` here? Or it does not matter 🤔

Reply:
CodeActions for taint issues are calculated separately. Here we are in `createResolveIssueCodeAction()` and we know the issue is not taint and similarly in `computeCodeActionsForTaintIssues()` we know the issue is always a taint issue.